### PR TITLE
Refactoring gyazo ruby script

### PR DESCRIPTION
**Pull request by drobune**
It seems getting complex. I try to refactor.

I generate GyazoClient class.The reasons is next.
- Easy to test.
- Put codes of environment variables in one.

Also I add brief test for GyazoClient.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
+
```
Comment (pastak) [inline Gemfile]: `group :development`

File: circle.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ machine:
 dependencies:
   pre:
     - gem install package_cloud
```
Comment (pastak) [inline circle.yml]: You can move this line to Gemfile

File: src/gyazo.rb [hunk 3]
```diff
@@ -1,130 +1,54 @@
...
-id = ''
-if File.exist?(idfile) then
-  id = File.read(idfile).chomp
-end
+config_file = "#{ENV['HOME']}/.gyazo.config.yml"
+opts = {}
+opts = YAML.load_file(config_file) if File.exist?(config_file)
 
-# get active window name
+gyazo_client = GyazoClient.new(opts)
```
Comment (pastak) [inline src/gyazo.rb]: [IMO] I think it's better that it passes yaml path to GyazoClient then YAML load in `GyazoClient`'s code

File: src/gyazo_client.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
+require 'net/http'
+require 'open3'
+require 'openssl'
+
+class GyazoClient
+
+  def initialize(opts = {})
+    default_opts = {
+      id_file:       ENV['HOME'] + "/.gyazotest.id",
```
Comment (pastak) [inline src/gyazo_client.rb]: test code?

File: src/gyazo.rb (hunk 3 above)
Comment (drobune) [inline src/gyazo.rb]: I think it is not good that `GyazoClient` class has relation with yaml file.
You can use any other format such like csv, plaintext, json, if `GyazoClient` accept hash.

File: src/gyazo_client.rb (hunk 4 above)
Comment (drobune) [inline src/gyazo_client.rb]: Sorry, this is my mistake. I leave debug codes.
I will fix  "/.gyazotest.id" -> "/.gyazo.id"

File: src/gyazo.rb (hunk 3 above)
Comment (pastak) [inline src/gyazo.rb]: I understand 😄  that's good way I think too.

**Comment by pastak** [comment]
[Gyazo](https://gyazo.com/6849bd29384e05b7c2b0fabef1d41de4)

(Maybe merge squash makes cleaner commit log)

File: src/gyazo_client.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    gyazo_url = res.response.body
+
+    new_id = res.response['X-Gyazo-Id']
+    save_id(new_id) if new_id && new_id != ''
+
+    gyazo_url
+  end
+
+  private
+  def gyazo_id
```
Comment (rakusai) [inline src/gyazo_client.rb]: should call this `device_id`?

Comment (drobune) [inline src/gyazo_client.rb]: Umm... I see. I will replace this method name to "device_id".

**Comment by nna774** [comment]
close too old

**Pull request closed without merging**